### PR TITLE
refactor: GUI error handling

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [x] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/plugins/error-reporting.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import * as Sentry from '@sentry/vue'
+import type { Api } from 'nocodb-sdk'
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nc-gui/plugins/error-reporting.ts:2]: all of these will auto-import

File: packages/nc-gui/plugins/error-reporting.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      } else {
+        errorReporting = null
+      }
+    },
+    { immediate: true },
+  )
+
+  function report(error: Error) {
+    console.log(error)
+
```
Comment (mertmit) [inline packages/nc-gui/plugins/error-reporting.ts:77]: Is this intended or leftover?

**Review by mertmit — APPROVED** [review]
LGTM - please just check the console log part

File: packages/nc-gui/plugins/error-reporting.ts (hunk 2 above)
Comment (pranavxc) [reply packages/nc-gui/plugins/error-reporting.ts:77]: Yes it's intentional, user can copy error from console and report if they disabled telemetry.

File: packages/nc-gui/nuxt-shim.d.ts [hunk 3]
```diff
@@ -12,6 +12,8 @@ declare module '#app' {
     }
     /** {@link import('./plugins/tele') Telemetry} Emit telemetry event */
     $e: (event: string, data?: any) => void
+    /** {@link import('./plugins/report') Error reporting} Error reporting */
+    $report: (event: Error) => void
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nc-gui/nuxt-shim.d.ts:16]: There is error-reporting plugin in which we are defining it `packages/nc-gui/plugins/error-reporting.ts`

**Review by dstala — APPROVED** [review]

**Pull request merged**